hudi-client/src/main/java/org/apache/hudi/client/AbstractHoodieWriteClient.java

```diff
@@ -186,10 +188,14 @@ public HoodieMetrics getMetrics() {
    * Get HoodieTable and init {@link Timer.Context}.
    *
    * @param operationType write operation type
+   * @param instantTime current inflight instant time
    * @return HoodieTable
    */
-  protected HoodieTable getTableAndInitCtx(WriteOperationType operationType) {
+  protected HoodieTable getTableAndInitCtx(WriteOperationType operationType, String instantTime) {
     HoodieTableMetaClient metaClient = createMetaClient(true);
+    if (config.shouldRollbackUsingMarkers()) {
```

Comment:
this was my thinking behind this guard. If someone wishes to stay with list based rollback, why execute this upgrade this which specifically does some work to assist in marker based rollback which will never be used since marked based rollback is not going to be used at all. I am not very strong on this though. But tests need some fixes though as I rely on creating commits and marker files using client at first, by disabling marker based rollback. If we remove this guard, then tests need to manually create all data files and marker files. I am not saying that as a reason to keep this guard, just saying we have some extra work to be done.